### PR TITLE
Add perpendicular check for lines

### PR DESCRIPTION
**Pull request by Sedictious**
This seems like an interesting project, please let me know if everything looks alright!

### CODE REVIEW

File: cirno/Geometry/LineLike.cs [hunk 1]
```diff
@@ -15,6 +15,10 @@ public abstract class LineLike : IShape {
             return (P1 - P2).Cross(other.P1 - other.P2) < tolerance;
         }
 
+        public bool Perpendicular(Line other, float tolerance = 0.0001f) {
```
Comment (20chan) [inline cirno/Geometry/LineLike.cs]: Parameter `LineLike other` will be better than `LineLike other`.

**Review by 20chan — CHANGES_REQUESTED** [review]
Thank you for PR so much. I commented one change.

File: cirno/Geometry/LineLike.cs (hunk 1 above)
Comment (Sedictious) [reply cirno/Geometry/LineLike.cs]: Done! I mainly did this because ``Parallel`` also accepted ``Line`` as parameter. Shall I change that too?

Comment (20chan) [reply cirno/Geometry/LineLike.cs]: Sure, it was my mistake. Thanks.

**Pull request merged**